prow/flagutil/github.go

```diff
@@ -49,9 +52,12 @@ func (o *GitHubOptions) AddFlagsWithoutDefaultGitHubTokenPath(fs *flag.FlagSet)
 }
 
 func (o *GitHubOptions) addFlags(wantDefaultGitHubTokenPath bool, fs *flag.FlagSet) {
+	fs.StringVar(&o.base, "github-base", github.DefaultBase, "GitHub's default base endpoint (may differ for enterprise)")
 	o.endpoint = NewStrings(github.DefaultAPIEndpoint)
 	fs.Var(&o.endpoint, "github-endpoint", "GitHub's API endpoint (may differ for enterprise).")
 	fs.StringVar(&o.graphqlEndpoint, "github-graphql-endpoint", github.DefaultGraphQLEndpoint, "GitHub GraphQL API endpoint (may differ for enterprise).")
+	fs.StringVar(&o.AuthEndpoint, "github-auth-endpoint", github.DefaultAuthEndpoint, "Github oauth authorization endpoint (may differ for enterprise).")
```

Comment:
Why do these need to be separate? Can't we ask for the host, then determine the api endpoints, etc?

Reply:
Correct, I think that's also what @fejta asked. I'll make the change.

Reply:
Then I have to make `o.base` to be `o.Base` so that is can be used in other packages.

Reply:
So I made a change to add `GitHubOAuthClient` to github options as a method.